From 77876495d05cf38db7b44aa6ebb495a550164780 Mon Sep 17 00:00:00 2001
From: Sam Day <me@samcday.com>
Date: Sun, 4 Oct 2020 22:23:53 +0200
Subject: [PATCH] [qubes] special case dom0 qubesdb connection

 * That is, connect to qubesdb.sock, rather than qubesdb.dom0.sock
---
 qubes/vm/__init__.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/qubes/vm/__init__.py b/qubes/vm/__init__.py
index 31e35b7a..808942e1 100644
--- a/qubes/vm/__init__.py
+++ b/qubes/vm/__init__.py
@@ -420,7 +420,9 @@ class BaseVM(qubes.PropertyHolder):
             self._qdb_connection_watch.close()
 
         import qubesdb  # pylint: disable=import-error
-        self._qdb_connection_watch = qubesdb.QubesDB(self.name)
+        self.log.debug('Opening QubesDB %s', self.name)
+        self._qdb_connection_watch = (qubesdb.QubesDB)(
+            *([self.name] if self.name != 'dom0' else []))
         if loop is None:
             loop = asyncio.get_event_loop()
         loop.add_reader(self._qdb_connection_watch.watch_fd(),
-- 
2.28.0

